### PR TITLE
fix links in learning path

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added, or deleted? (Required)

Fix links in learning path.
Add learning path in TOC.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v1.4 (TiDB Operator 1.4 versions)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

**Review by qiancai — APPROVED** [review]

File: en/_index.md [hunk 1]
```diff
@@ -9,73 +9,73 @@ hide_commit: true
 
 <LearningPath label="Learn" icon="cloud1">
 
-[TiDB Operator Overview](https://docs.pingcap.com/tidb-in-kubernetes/dev/tidb-operator-overview)
+[TiDB Operator Overview](https://docs.pingcap.com/tidb-in-kubernetes/v1.4/tidb-operator-overview)
 
-[TiDB Operator Architecture](https://docs.pingcap.com/tidb-in-kubernetes/dev/architecture)
+[TiDB Operator Architecture](https://docs.pingcap.com/tidb-in-kubernetes/v1.4/architecture)
 
 [What's New in v1.3](https://docs.pingcap.com/tidb-in-kubernetes/dev/whats-new-in-v1.3)
```
Comment (ran-huang) [inline en/_index.md:16]: 这个要不先删了吧，1.4 里看到 what's new in 1.3 有点奇怪

File: zh/_index.md [hunk 2]
```diff
@@ -9,73 +9,73 @@ hide_commit: true
 
 <LearningPath label="了解" icon="cloud1">
 
-[TiDB Operator 简介](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/tidb-operator-overview)
+[TiDB Operator 简介](https://docs.pingcap.com/zh/tidb-in-kubernetes/v1.4/tidb-operator-overview)
 
-[TiDB Operator 架构](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/architecture)
+[TiDB Operator 架构](https://docs.pingcap.com/zh/tidb-in-kubernetes/v1.4/architecture)
 
 [TiDB Operator v1.3 新特性](https://docs.pingcap.com/zh/tidb-in-kubernetes/dev/whats-new-in-v1.3)
```
Comment (ran-huang) [inline zh/_index.md:16]: 同上

File: en/_index.md (hunk 1 above)
Comment (qiancai) [reply en/_index.md:16]: +1

**Review by ran-huang — APPROVED** [review]
rest LGTM

File: en/_index.md (hunk 1 above)
Comment (hfxsd) [reply en/_index.md]: 好

**Pull request merged**